perch/model.py

```diff
@@ -303,8 +378,20 @@ def can_create(cls, user, **data):
         raise Return(True)
 
     @coroutine
+    def can_approve(self, user, **data):
+        """Check if a user is authorized to approve a resource"""
+        raise Return(False)
+
+    @coroutine
     def can_update(self, user, **data):
         """Check if a user is authorized to update a resource"""
+
+        # Should only be able to edit resource if in editable state or being changed to editable state
+        current_state_editable = self.state in self.editable_states
+        new_state_editable = 'state' in data and getattr(State, data['state'], None) in self.editable_states
```

Comment:
Shouldn't `getattr(State, data['state'], None)` have `self.state` as the default? Otherwise would the user be able to update if the state is not changing?

I also think using `set.issubset` would be simpler, e.g.:

```python
states = {self.state, getattr(State, data.get('state'), self.state)}
can_update = states.issubset(set(self.editable_states))
raise Return((can_update, []))
```